pipeline_dp/pipeline_operations.py

```diff
@@ -86,6 +90,50 @@ def group_by_key(self, col, stage_name: str):
     def filter(self, col, fn, stage_name: str):
         return col | stage_name >> beam.Filter(fn)
 
+    def filter_by_key(self, col, public_partitions, data_extractors, stage_name: str):
+        """Filter out the partitions that are not not meant to be public.
+
+        Args:
+          col: input collection
+          public_partitions: collection of public partition keys
+          stage_name: name of the stage
+
+        Returns:
+          A filtered collection containing only data belonging to public_partitions
+
+        """
+        class PartitionsFilterJoin(beam.DoFn):
+          def process(self, joined_data):
+            key, rest = joined_data
+            values, is_public = rest.get('values'), rest.get('is_public')
+
+            # TODO the Issue #4 says this is blocked on other tasks. Revisit
+            # this once unblocked
+            if not values:
+              values = [None]
```

Comment:
Please replace this line with "return" (returning [None] would break further stages which are in development, later this will be fixed.

Reply:
Done.